bpf/bpf_host.c

```diff
@@ -1015,6 +1016,16 @@ int to_netdev(struct __ctx_buff *ctx __maybe_unused)
 	}
 #endif
 
+#if defined(ENABLE_EGRESS_GATEWAY)
+	if ((ctx->mark & MARK_MAGIC_SNAT_DONE) != MARK_MAGIC_SNAT_DONE) {
+		ret = egress_nat_fwd(ctx);
```

Comment:
> The easiest way to fix this would probably be to redirect() the packets to the native device as they arrive via the tunnel. That'll require a FIB lookup to select the device however. 

This is a good suggestion. This should be general to all the packets, not just for egress, right?

Reply:
I am not too worried about the iptables rules inserted by IP masq agent, since cilium will have it implemented in epbf as well. But I agree, we should skip kernel stack as much as possible.

Reply:
> This is a good suggestion. This should be general to all the packets, not just for egress, right?

Yeah, I think so, at least to avoid asymmetric paths.